src/manifests_cert_manager.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2024 Ubuntu
+# See LICENSE file for licensing details.
+
+"""Implementation of cert-manager kubernetes manifests."""
+
+import logging
+import pickle
+from hashlib import md5
+from typing import Dict
+
+from ops.manifests import ConfigRegistry, ManifestLabel, Manifests, Patch
+
+log = logging.getLogger()
+
+
+class RemoveNamespace(Patch):
+    """Remove namespace from resources so they deploy to the charm's namespace."""
+
+    def __call__(self, obj):
+        """Remove namespace from resources."""
+        obj.metadata.namespace = None
+
+
+class CertManagerManifests(Manifests):
+    """Deployment details for cert-manager."""
+
+    def __init__(self, charm, charm_config):
+        super().__init__(
+            "cert-manager",
+            charm.model,
+            "upstream/cert-manager",
+            [
+                ManifestLabel(self),
+                ConfigRegistry(self),
```

Comment:
I'm betting this one isn't necessary.  The point of `ConfigRegistry` is to alter any container images based on a manifest config called `image-registry`.  I don't see that in your `charm_config` so you can likely drop it.  I'm betting you are expecting to not change the source repositories for the images?

https://github.com/canonical/ops-lib-manifest/blob/main/ops/manifests/manipulations.py#L203

Reply:
Yup, you're right. I dropped `ConfigRegistry` here and in the other manifests class.